lib/file-promise.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable no-undef */
+'use strict';
+
+const fs = require('fs');
+const util = require('util');
+/** promisfay read and write fs module */
+const write = util.promisify(fs.writeFile);
+const read = util.promisify(fs.readFile);
+
+module.exports = exports = {};
+/**
+ * wraps fs.readFile as a promise, process the file as a JSON string, resolving with JS object or null
+ * Rejects with error on all fs errors as well as a malformed JSON string in the file itself
+ * @param file - Full filesystem path the file to read
+ * @returns {Promise<T>}
+ * */
+exports.read = (file) =>
+  read(file)
+    .then(contents => {
+      try {
+        return JSON.parse(contents.toString().trim());
+      } catch (e) {
+        throw 'Invalid JSON';
+      }
+    })
+    .catch(err => {
+      throw err;
+    });
```

Comment:
This is the proper way to `catch` errors when using Promises. You can use `try`/`catch` blocks when you're using `async`/`await`